cmake/modules/FindDav1d.cmake

```diff
@@ -0,0 +1,70 @@
+#.rst:
+# FindDav1d
+# --------
+# Finds the dav1d library
+#
+# This will define the following variables::
+#
+# DAV1D_FOUND - system has dav1d
+# DAV1D_INCLUDE_DIRS - the dav1d include directory
+# DAV1D_LIBRARIES - the dav1d libraries
+#
+
+# required dav1d library versions
+set(REQUIRED_DAV1D_VERSION 0.5)
+
+# TODO Detect existing dav1d
+
+message(STATUS "dav1d ${REQUIRED_DAV1D_VERSION} not found, falling back to internal build")
+
+# Internal dav1d
+if(NOT DAV1D_FOUND)
+  include(ExternalProject)
+
+  # Extract version
+  file(STRINGS ${CMAKE_SOURCE_DIR}/tools/depends/target/dav1d/DAV1D-VERSION VER)
+  string(REGEX MATCH "VERSION=[^ ]*$.*" DAV1D_VER "${VER}")
+  list(GET DAV1D_VER 0 DAV1D_VER)
+  string(SUBSTRING "${DAV1D_VER}" 8 -1 DAV1D_VER)
+
+  # Extract repository URL
+  string(REGEX MATCH "BASE_URL=([^ ]*)" DAV1D_BASE_URL "${VER}")
+  list(GET DAV1D_BASE_URL 0 DAV1D_BASE_URL)
+  string(SUBSTRING "${DAV1D_BASE_URL}" 9 -1 DAV1D_BASE_URL)
+
+  # allow user to override the download URL with a local tarball
+  # needed for offline build envs
+  if(DAV1D_URL)
+    get_filename_component(DAV1D_URL "${DAV1D_URL}" ABSOLUTE)
+  else()
+    set(DAV1D_URL ${DAV1D_BASE_URL}/archive/${DAV1D_VER}.tar.gz)
+  endif()
+
+  if(VERBOSE)
+    message(STATUS "DAV1D_URL: ${DAV1D_URL}")
+  endif()
+
+  set(DAV1D_LIBRARY ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/lib/dav1d.a)
+  set(DAV1D_INCLUDE_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/include)
+
+  externalproject_add(dav1d
+                      URL ${DAV1D_URL}
+                      DOWNLOAD_NAME dav1d-${DAV1D_VER}.tar.gz
+                      DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/download
+                      UPDATE_COMMAND ""
+                      CONFIGURE_COMMAND ${CMAKE_CURRENT_BINARY_DIR} meson <BINARY_DIR> <SOURCE_DIR>
+                            --buildtype=release
+                            --default-library=static
+                            --prefix=${CORE_BUILD_DIR}/dav1d
+                      BUILD_COMMAND ${CMAKE_CURRENT_BINARY_DIR} ninja -C <BINARY_DIR>
+                      INSTALL_COMMAND ${CMAKE_CURRENT_BINARY_DIR} ninja -C <BINARY_DIR> install
+                      BUILD_BYPRODUCTS ${DAV1D_LIBRARY})
```

Comment:
This is not working. dav1d is using ninja for building. I don't know how to correctly invoke a meson/ninja build via cmake... 

Would be nice if someone could help me out.

Reply:
You should look at how it's done for flatbuffers, crossguid, etc to allow using the `ENABLE_INTERNAL_LIBDAV1D` style logic. See, https://github.com/xbmc/xbmc/blob/master/CMakeLists.txt#L54-L58

Otherwise for the cmake invoking meson/ninja I don't know of any other packages that use that. I think you are mostly correct though by using `externalproject_add`.

Reply:
Thanks, got it working now. Still trying to figure out how I can tell ffmpeg where to find the installed dav1d-lib.